trieste/models/interfaces.py

```diff
@@ -226,3 +249,91 @@ def optimize(self, dataset: Dataset) -> None:
 
         for model, obs in zip(self._models, observations):
             model.optimize(Dataset(dataset.query_points, obs))
+
+
+
+    def reparam_sampler(self, num_samples: int) -> ReparametrizationSampler:
+        """
+        Return a reparametrization sampler providing `num_samples` samples.
+
+        Note that this is not supported by all models.
+
+        :param num_samples: The desired number of samples.
+        :return: The reparametrization sampler.
+        """
+        raise NotImplementedError(f"Model {self!r} does not have a reparametrization sampler")
+
+
+
+class ReparametrizationSampler(ABC):
+    r"""
+    This sampler employs the *reparameterization trick* to draw samples from a
+    :class:`ProbabilisticModel`\ 's predictive distribution  across a discrete set of
+    points.
+    """
+
+    def __init__(self, sample_size: int, model: ProbabilisticModel):
+        """
+        :param sample_size: The desired number of samples.
+        :param model: The model to sample from.
+        :raise ValueError (or InvalidArgumentError): If ``sample_size`` is not positive.
+        """
+        tf.debugging.assert_positive(sample_size)
+
+        self._sample_size = sample_size
+        self._model = model
+        self._initialized = tf.Variable(False)
+
+    def __repr__(self) -> str:
+        """"""
+        return f"{self.__class__.__name__}({self._sample_size!r}, {self._model!r})"
+
+    @abstractmethod
+    def sample(self, at: TensorType, *, jitter: float = DEFAULTS.JITTER) -> TensorType:
+        """
+        :param at: Input points that define the sampler.
+        :param jitter: The size of the jitter to use when stabilising the Cholesky
+            decomposition of the covariance matrix.
+        :return: Samples.
```

Comment:
It's not clear to me why we want jitter here. Most of the reparameterization samplers will end up calling the model's `predict` method to get `mean` and `var` with `full_cov=False`, so any need for jitter should be the model's responsibility

Reply:
jitter has always been something that we specify when calling the sampler. But I agree with you @sebastianober. WHat do you think @vpicheny ?

Reply:
no strong opinion, but we need to make sure that we can tweak those jitters when needed, they sometimes have a too strong influence on the results.